cpr/session.cpp

```diff
@@ -558,6 +563,14 @@ void Session::Impl::SetSslOptions(const SslOptions& options) {
     if (!options.ca_path.empty()) {
         curl_easy_setopt(curl_->handle, CURLOPT_CAPATH, options.ca_path.c_str());
     }
+#if SUPPORT_CURLOPT_SSL_CTX_FUNCTION
+#ifdef OPENSSL_BACKEND_USED
```

Comment:
In case the `OPENSSL_BACKEND_USED` is not used, please throw an error explaining why OpenSSL is required.
This might help:
https://gcc.gnu.org/onlinedocs/cpp/Diagnostics.html

Reply:
I think I don't understand exactly what you mean.
As I understand it, it would look something like this:
```c++
#if SUPPORT_CURLOPT_SSL_CTX_FUNCTION
#ifdef OPENSSL_BACKEND_USED
...
#else
#error "CURLOPT_SSL_CTX_FUNCTION is only supported for OpenSSL"
#endif
```
However, this would mean that one could no longer compile CPR without OpenSSL.

Reply:
Ah, did not fully get the `SUPPORT_CURLOPT_SSL_CTX_FUNCTION`. Then it is fine!